flask_apispec/wrapper.py

```diff
@@ -53,14 +53,17 @@ def marshal_result(self, unpacked, status_code):
         annotation = utils.resolve_annotations(self.func, 'schemas', self.instance)
         schemas = utils.merge_recursive(annotation.options)
         schema = schemas.get(status_code, schemas.get('default'))
-        if schema and annotation.apply is not False:
+        if schema and schema['schema'] and annotation.apply is not False:
             schema = utils.resolve_schema(schema['schema'], request=flask.request)
             dumped = schema.dump(unpacked[0])
             output = dumped.data if MARSHMALLOW_VERSION_INFO[0] < 3 else dumped
         else:
             output = unpacked[0]
+        if not output and status_code == 204:
```

Comment:
Why is it necessary to check the status code?

Reply:
The reason is I got a webtest lint error when passing 204 response through `format_response` due to `Content-Type` header being added.   
But I think that's fine for other status codes, the result would just be a JSON `null` or something.